### PR TITLE
[#148] Add custom multipart reader

### LINKED ISSUE #148: Investigate the usage of bigger body chunks in multipart upload
Multipart reader drastically slows down upload because it reads data with small [4 KiB chunks](https://github.com/golang/go/blob/91b9915d3f6f8cd2e9e9fda63f67772803adfa03/src/mime/multipart/multipart.go#L32), while we can upload data to NeoFS with up to [3 MiB chunks](https://github.com/nspcc-dev/neofs-sdk-go/blob/2deaaeef05027f6ed52f273c613cab52ecf764c0/pool/pool.go#L250). 

Increasing chunk size can increase upload speed up to 400% in naive tests, so we can dig more into it for special high-performance data upload cases.

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #148

### CODE REVIEW

**Comment by KirillovDenis** [comment]
go benchmark:
```
BenchmarkAll/bare
BenchmarkAll/bare-16         	      10	 104067101 ns/op
BenchmarkAll/default
BenchmarkAll/default-16      	       4	 292874467 ns/op
BenchmarkAll/custom
BenchmarkAll/custom-16       	       6	 176841604 ns/op
```

k6 benchmark:

```
default multipart reader
--
time: 01m00s
data_received..................: 2.5 kB 42 B/s
data_sent......................: 1.3 GB 22 MB/s

custom multipart reader
--
time: 03m39s
data_received..................: 2.5 kB 12 B/s
data_sent......................: 1.3 GB 6.2 MB/s

```

**Comment by KirillovDenis** [comment]
We should wait for https://github.com/nspcc-dev/neofs-http-gw/pull/161

File: uploader/multipart/multipart.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+	"net/textproto"
+	"strings"
+)
+
+var emptyParams = make(map[string]string)
+
+// This constant needs to be at least 76 for this package to work correctly.
+// This is because \r\n--separator_of_len_70- would fill the buffer and it
+// wouldn't be safe to consume a single byte from it.
+const peekBufferSize = 3 << 20
```
Comment (alexvanin) [inline uploader/multipart/multipart.go:31]: So this is the buffer size we are changing? If so, then small comment will be appreciated there too.

**Review by alexvanin — DISMISSED** [review]
LGTM. You can rebase and merge.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**